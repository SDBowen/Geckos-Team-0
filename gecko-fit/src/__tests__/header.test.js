import React from 'react';
import { shallow, render} from 'enzyme';
import Header from '../components/header';

jest.useFakeTimers();
// test Header render()
describe('Header Component', () => {
  it('Should render header correctly', () => {
    const wrapper = shallow(<Header />);
    const header = {};

    expect(
      wrapper.exists(
        <header className="App__header" style={header}>
        </header>
      )
    ).toBe(true);
  });
  // test componentDidMount calling setInterval and setInterval calling randomImg
  it('timer that will call randomImg function fires', () => {
    const wrapper = shallow(<Header />);
    // creates a spy to spy on Header prototype randomImg
    const spy = jest.spyOn(Header.prototype, 'randomImg');
    // creates instance of randomImg
    wrapper.instance().randomImg();
    expect(setInterval).toHaveBeenCalled();
    expect(spy).toHaveBeenCalled();
    
  });
});