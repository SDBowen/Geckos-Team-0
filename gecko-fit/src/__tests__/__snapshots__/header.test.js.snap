// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render header correctly 1`] = `
<header
  className="App__header"
  style={
    Object {
      "backgroundColor": "black",
      "backgroundImage": "url(\\"Food_00.jpg\\")",
      "backgroundPosition": "center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "100% auto",
    }
  }
>
  <h1
    className="App__title"
  >
    Gecko Fit
  </h1>
</header>
`;
